specs/interop.md

```diff
@@ -0,0 +1,188 @@
+a strict synchrony assumption on the sequencer that it MUST have all unsafe blocks of remote chains
+as fast as possible to ensure that it is building correct blocks. It also prevents a class of attacks
+where the user can send a relaying message before the initiating message and trick the derivation pipeline
+into reorganizing the sequencer.
+
+### Safety
+
+The initiating messages for all relaying messages MUST be resolved as safe before an L2 block can transition from being unsafe to safe.
+Users MAY optimistically accept unsafe blocks without any verification of the relaying messages. They SHOULD optimistically verify
+the initiating messages exist in remote unsafe blocks to more quickly reorganize out invalid blocks.
+
+## Predeploys
+
+Two new system level predeploys are introduced for managing cross chain messaging.
+
+### CrossL2Outbox
+
+Address: `0x4200000000000000000000000000000000000023`
+
+The `CrossL2Outbox` is responsible for initiating a cross chain message on the local chain.
+For every cross chain message that is sent, it MUST emit an event that indicates the existence
+of the cross chain message.
+
+An initiating message 
+
+### CrossL2Inbox
+
+Address: `0x4200000000000000000000000000000000000022`
+
+The `CrossL2Inbox` is responsible for relaying a cross chain message on the remote chain.
+It is permissionless to finalize a cross chain message on behalf of any user. Certain protocol
+enforced invariants must be preserved to ensure safety of the protocol.
+
+### Invariants
+
+- The timestamp of relaying message MUST be greater than or equal to the timestamp of the initiating message
+- The chain id of the initiating message MUST be in the interop set
+- The relaying message MUST be initiated by an externally owned account such that the top level EVM call frame enters the `CrossL2Inbox`
+
+#### Only EOA Invariant
+
+The `onlyEOA` invariant on relaying a cross chain message enables static analysis on relaying messages.
+This allows for the derivation pipeline and block builders to reject relaying messages that do not
+have a corresponding initiating message.
+
+### L1Block
+
+The `L1Block` contract is updated to include the set of allowed chains. The L1 Atrributes transaction
+sets the set of allowed chains. The `L1Block` contract MUST provide a public getter to check is a particular
+chain is in the interop set.
+
+The `setL1BlockValuesInterop()` function MUST be called on every block after the interop upgrade block.
+The interop upgrade block itself MUST include a call to `setL1BlockValuesEcotone`.
+
+## SystemConfig
+
+The `SystemConfig` is updated to manage the interop set. The chain operator can add or remove chains from the interop set
+through the `SystemConfig`. A new `ConfigUpdate` event `UpdateType` enum is added that corresponds to a change in the interop set.
+
+The `SystemConfig` MUST enforce that the maximum size of the interop set is `type(uint8).max` or 255.
```

Comment:
Is this enough?

Reply:
until we have zk, this 255 feels like a nice upper bound the set given the scaling constraints

Reply:
I added in a section under security concerns that talks about why this value was chosen. There are 2 reasons that are both security related